File: Caelan.Frameworks.Common.NUnit/PasswordTest.cs
using NUnit.Framework;
using System;
using System.Diagnostics;
using Caelan.Frameworks.Common.Classes;
using System.Runtime.CompilerServices;

namespace Caelan.Frameworks.Common.NUnit
{
	[TestFixture]
	public class PasswordTest
	{
		public class PasswordManager : PasswordHasher
		{
			public PasswordManager() : base("salt", "default")
			{
			}
		}

		[Test]
		public void TestPassword()
		{
			var pwd = new PasswordManager();

			Console.WriteLine(pwd.HashPassword("password"));
		}
	}
}

File: Caelan.Frameworks.Common.NUnit/PasswordTest.fs
﻿namespace Caelan.Frameworks.Common.NUnit

open System
open NUnit.Framework
open Caelan.Frameworks.Common.Classes
open Caelan.Frameworks.PasswordHashing.Classes

type CustomPasswordHasher() = 
    inherit PasswordHasher("salt", "default")

[<TestFixture>]
type PasswordTest() = 
    [<Test>]
    member __.TestHasher() = 
        let pwd = CustomPasswordHasher()
        "password"
        |> pwd.HashPassword
        |> printfn "%s"
        pwd.DefaultPasswordHashed |> printfn "%s"
